docs/plugins/discovery-ec2.asciidoc

```diff
@@ -1,167 +1,215 @@
-http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/Using_Tags.html[Read more about EC2 Tags].
+Each EC2 instance has a number of different addresses that might be suitable
+for discovery. This setting allows you to select which of these addresses is
+used by the discovery process. It can be set to one of `private_ip`,
+`public_ip`, `private_dns`, `public_dns` or `tag:TAGNAME` where `TAGNAME`
+refers to a name of a tag. This setting defaults to `private_ip`.
 
-Defaults to `private_ip`.
---
+If you set `discovery.ec2.host_type` to a value of the form `tag:TAGNAME` then
+the value of the tag `TAGNAME` attached to each instance will be used as that
+instance's address for discovery. Instances which do not have this tag set will
+be ignored by the discovery process.
 
-`availability_zones`::
+For example if you tag some EC2 instances with a tag called
+`my-elasticsearch-host` and set `host_type: tag:my-elasticsearch-host` then the
+`discovery-ec2` plugin will read each instance's host name from the
+`my-elasticsearch-host` tag.
+http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/Using_Tags.html[Read more
+about EC2 Tags].
 
-    Either a comma separated list or array based list of availability zones.
-    Only instances within the provided availability zones will be used in the
-    cluster discovery.
+--
 
-`any_group`::
+`discovery.ec2.availability_zones`::
 
-    If set to `false`, will require all security groups to be present for the
-    instance to be used for the discovery. Defaults to `true`.
+    A list of the names of the availability zones to use for discovery. Only
+    instances placed in one of the given availability zones will be used for
+    discovery.
 
-`node_cache_time`::
+[[discovery-ec2-filtering]]
+`discovery.ec2.tag.TAGNAME`::
+
++
+--
 
-    How long the list of hosts is cached to prevent further requests to the AWS API.
-    Defaults to `10s`.
+A list of the values of a tag called `TAGNAME` to use for discovery. If set,
+only instances that are tagged with one of the given values will be used for
+discovery. For instance, the following settings will only use nodes with a
+`role` tag set to `master` and an `environment` tag set to either `dev` or
+`staging`.
 
-*All* secure settings of this plugin are {ref}/secure-settings.html#reloadable-secure-settings[reloadable].
-After you reload the settings, an aws sdk client with the latest settings
-from the keystore will be used.
+[source,yaml]
+----
+discovery.ec2.tags.role: master
+discovery.ec2.tags.environment: dev,staging
+----
 
-[IMPORTANT]
-.Binding the network host
-==============================================
+One practical use for tag filtering is when an EC2 cluster contains many nodes
+that are not master-eligible {es} nodes. In this case, tagging the EC2 instances
+that _are_ running the master-eligible {es} nodes, and then filtering by that
+tag, will help discovery to run more efficiently.
```

Comment:
I wish this advice was not buried in the settings section here. Perhaps we can make note at the intro section that in general we want to find the master-eligible nodes and that it helps if this subset of nodes can be directly identified using the EC2 service endpoint.

Reply:
Ok, I adjusted the wording in d0ccbbd803b5a03460712b2d4bc575526eaf0cde.